source/slang/slang-mangle.cpp

```diff
@@ -474,5 +474,15 @@ namespace Slang
         return context.sb.ProduceString();
     }
 
+    String getHashedName(const UnownedStringSlice& mangledName)
+    {
+        uint64_t hash = GetHashCode64(mangledName.begin(), mangledName.size());
```

Comment:
Eventually we would need to lock down and document what hash function is used to generate our hashed symbol names.

For right now, when the mangled names that are input to this step aren't really stable or carefully designed, any attempt to document things would be a bit silly. The caveat at this point is that all the inputs would need to have been created by the same version of the compiler (and we should obviously treat breaking changes to mangling as backwards-incompatible changes now that serialization is being exposed to users).